e2e-tests/src/balance.rs

```diff
@@ -0,0 +1,216 @@
+use anyhow::{anyhow, Result};
+use ethers::types::U256;
+
+/// Struct holding the current balances on both chains for a given account pair.
+/// It includes the native coins, and all the wrapped tokens we're concerned with.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Balance {
```

Comment:
:+1: nice. I'd add a comment that the impl serves as a model for the transfers we care about.

Reply:
done